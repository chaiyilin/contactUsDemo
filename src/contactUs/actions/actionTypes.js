export const UPDATE_AD_AMOUNT = 'UPDATE_AD_AMOUNT';
export const AUTH_LOGIN = 'AUTH_LOGIN';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';