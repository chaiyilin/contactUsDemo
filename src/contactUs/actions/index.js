import {
    UPDATE_AD_AMOUNT,
    AUTH_LOGIN,
    AUTH_LOGOUT
} from './actionTypes';

export const updateAdAmount = (productId, amount) => {
    return {
        type: UPDATE_AD_AMOUNT,
        payload: {
            productId,
            amount
        }
    };
};

export const login = (username, password) => {
    return {
        type: AUTH_LOGIN,
        payload: {
            username,
            password
        }
    };
};

export const logout = () => {
    return {
        type: AUTH_LOGOUT,
        payload: {
        }
    };
};