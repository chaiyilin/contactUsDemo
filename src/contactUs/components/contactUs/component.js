import React, {Component} from 'react'
import {connect} from 'react-redux'
import {Field, reduxForm} from 'redux-form'
import Paper from 'material-ui/Paper'
import RaisedButton from 'material-ui/RaisedButton'
import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin()

import {contactUsFormContainerStyle, contactUsFormStyle, fieldsContainerStyle, buttonStyle} from './style'
import {required, maxLength20, maxLength200, email} from './validation'
import textFieldRender from './textFieldRender'
import {submit} from '../../actions'


class ContactUs extends Component {

    constructor(props) {
        super(props)
    }

    render() {
        const {handleSubmit, pristine, reset, submitting} = this.props
        return (
            <div className={`${contactUsFormContainerStyle}`}>
                <Paper className={`${contactUsFormStyle}`} zDepth={3}>
                    <form onSubmit={handleSubmit}>
                        <div className={`${fieldsContainerStyle}`}>
                            <Field name="name" component={textFieldRender} validate={[required, maxLength20]}
                                   floatingLabelText="Name *"/>
                            <Field name="email" component={textFieldRender} validate={[required, email]}
                                   floatingLabelText="Email *"/>
                            <Field name="message" component={textFieldRender} validate={[required, maxLength200]}
                                   multiLine={true} floatingLabelText="Message *"/>
                            <RaisedButton
                                className={`${buttonStyle}`}
                                type="submit"
                                label='Submit'
                                disabled={submitting}
                                labelPosition={'before'}
                                fullWidth={true}
                                primary={true}
                            />
                        </div>
                    </form>
                </Paper>
            </div>
        )
    }
}
export default connect(null, {submit})(reduxForm({form: 'ContactUs'})(ContactUs));
//export default reduxForm({form: 'ContactUs'})(ContactUs);