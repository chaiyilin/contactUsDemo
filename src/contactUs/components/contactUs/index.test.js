import React from 'react'
import {ContactUs} from './component'
import {submit} from '../../actions'

//Testing basic component rendering
it('should render ContactUs', () => {
  const wrapper = shallow(
    <ContactUs/>
  );
  expect(wrapper).toMatchSnapshot();
});

//Testing props
/*it('should render a document title', () => {
  const wrapper = shallow(
    <ContactUs onSubmit={submit} />
  );
  expect(wrapper.prop('onSubmit')).toEqual('submit');
});*/
