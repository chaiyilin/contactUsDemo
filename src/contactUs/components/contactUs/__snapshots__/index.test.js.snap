// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContactUs 1`] = `
<div
  className="css-1ir39kl"
>
  <Paper
    circle={false}
    className="css-1be6uqv"
    rounded={true}
    transitionEnabled={true}
    zDepth={3}
  >
    <form>
      <div
        className="css-v9t76d"
      >
        <Field
          component={[Function]}
          floatingLabelText="Name *"
          name="name"
          validate={
            Array [
              [Function],
              [Function],
            ]
          }
        />
        <Field
          component={[Function]}
          floatingLabelText="Email *"
          name="email"
          validate={
            Array [
              [Function],
              [Function],
            ]
          }
        />
        <Field
          component={[Function]}
          floatingLabelText="Message *"
          multiLine={true}
          name="message"
          validate={
            Array [
              [Function],
              [Function],
            ]
          }
        />
        <RaisedButton
          className="css-a3bf9n"
          disabled={false}
          fullWidth={true}
          label="Submit"
          labelPosition="before"
          primary={true}
          secondary={false}
          type="submit"
        />
      </div>
    </form>
  </Paper>
</div>
`;
